mmdet/models/backbones/xception.py

```diff
@@ -0,0 +1,317 @@
+                    conv_cfg,
+                    filters,
+                    filters,
+                    3,
+                    stride=1,
+                    padding=1,
+                    bias=False))
+            if self.batch_norm:
+                rep.append(build_norm_layer(norm_cfg, filters))
+
+        if not grow_first:
+            rep.append(self.relu)
+            rep.append(
+                SeparableConv(
+                    conv_cfg,
+                    in_filters,
+                    out_filters,
+                    3,
+                    stride=1,
+                    padding=1,
+                    bias=False))
+            if self.batch_norm:
+                rep.append(build_norm_layer(norm_cfg, out_filters))
+
+        if not start_with_relu:
+            rep = rep[1:]
+        else:
+            rep[0] = nn.ReLU(inplace=False)
+
+        if stride != 1:
+            if conv_cfg['type'] == 'Conv3d':
+                rep.append(nn.MaxPool3d(3, stride, 1))
+            elif conv_cfg['type'] == 'Conv1d':
+                rep.append(nn.MaxPool1d(3, stride, 1))
+            else:
+                rep.append(nn.MaxPool2d(3, stride, 1))
+        self.rep = nn.Sequential(*rep)
+
+    def forward(self, inp):
+        x = self.rep(inp)
+
+        if self.skip is not None:
+            skip = self.side(inp)
+            if self.batch_norm:
+                skip = self.bn(skip)
+        else:
+            skip = inp
+
+        x += skip
+        return x
+
+
+@BACKBONES.register_module()
+class Xception(nn.Module):
+    """Xception optimized for the ImageNet dataset, as specified in
+    https://arxiv.org/pdf/1610.02357.pdf."""
+
+    def __init__(self,
+                 depth=8,
+                 conv_cfg=dict(type='Conv2d'),
+                 norm_cfg=dict(type='BN'),
+                 conv_size=[32, 64, 128, 256, 728, 1024, 1536, 2048]):
+        """ Constructor
+        Args:
+            depth: Number of middle units in the backbone
+        """
+        super(Xception, self).__init__()
+
+        self.norm_cfg = norm_cfg
+        self.conv_cfg = conv_cfg
+        self.depth = depth
+        self.conv_sizes = conv_size
+        self.bn = False if norm_cfg is None else len(norm_cfg) != 0
+        self.conv1 = build_conv_layer(
+            self.conv_cfg, 3, self.conv_sizes[0], 3, 2, 0, bias=False)
+        if self.bn:
+            self.bn1_name, self.bn1 = build_norm_layer(self.norm_cfg,
+                                                       self.conv1.out_channels)
+        self.relu = nn.ReLU(inplace=True)
+
+        self.conv2 = build_conv_layer(
+            self.conv_cfg,
+            self.conv1.out_channels,
+            self.conv_sizes[1],
+            3,
+            bias=False)
+        if self.bn:
+            self.bn2_name, self.bn2 = build_norm_layer(self.norm_cfg,
+                                                       self.conv2.out_channels)
+        # do relu here
+
+        self.block1 = Block(
```

Comment:
1. Use `self.layer1`, `self.layer2`, which are more consistent with ResNets.
2. Can initialize the blocks/stage layers in a more general form as [that in ResNet](https://github.com/open-mmlab/mmdetection/blob/master/mmdet/models/backbones/resnet.py#L421)?